packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -99,6 +130,98 @@ export const getKey = (node: t.Expression) => {
   throw new Error();
 };
 
+/**
+ * Returns `true` if an identifier or any paths that reference the identifier are mutated.
+ * @param path
+ */
+const isIdentifierReferencesMutated = (path: NodePath<t.Identifier>): boolean => {
+  const binding = path.scope.getBinding(path.node.name);
+  if (!binding) {
+    return false;
+  }
+
+  if (!t.isVariableDeclarator(binding.path.node) || !binding.constant) {
+    return true;
+  }
+
+  for (let i = 0; i < binding.referencePaths.length; i++) {
+    const refPath = binding.referencePaths[i];
+    const innerBinding = refPath.scope.getOwnBinding(path.node.name);
```

Comment:
i have to wonder if this should be getBinding..

Reply:
yeah still passes, pushing change

Reply:
yeah..`getBinding` is better. It will traverse more right. We can have test cases for this if we want.

Reply:
any link for the differences?

Reply:
https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#scope

See this

Reply:
dont really see anything going in depth around the get binding functions though 🤔

Reply:
tests tests tests..they might help..!! 😛

Reply:
May be something like this

```typescript
function cube(n) {
  function square(n) {
    return n * n;
  }

  return n * square(n);
}
```

Reply:
if this ends up being called as `cube(2)` it would just bail out to evaluate it as a dynamic variable :)